### PR TITLE
Add some market endpoints?

### PR DESCRIPTION
**Pull request by maurerit**
This adds the Market order endpoints, both the Bulk and non Bulk calls.  Made a minor change in how the path type is referenced... not entirely sure I like it but it seems clean enough and won't hard code the hostname at least.

### CODE REVIEW

File: retrofit/src/main/java/org/devfleet/crest/retrofit/CrestServiceImpl.java [hunk 1]
```diff
@@ -269,6 +276,64 @@ private boolean setWaypoints(final List<CrestWaypoint> waypoints, final boolean
...
+            } while (dictionary.getPageCount() > page);
+            return returned;
+        } catch (IOException e) {
+            LOG.error(e.getLocalizedMessage(), e);
+            throw new IllegalStateException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    @Override
+    public void getAllOrdersAsync(final long regionId, final Callback<CrestDictionary<CrestMarketBulkOrder>> cb) {
```
Comment (evanova) [inline retrofit/src/main/java/org/devfleet/crest/retrofit/CrestServiceImpl.java]: I don't really want this layer to provide async features. The callers of the interface should do it if they feel like it.

The way the interface is typically used is using RxJava, which will do a better job of managing threads. But again, clients of the interface should deal with async, not us. 

Please remove this method

**Comment by evanova** [comment]
Besides this async thing, it looks good. Thanks

**Comment by maurerit** [comment]
Async method removed :D.  Wasn't sure how you wanted to go about things such as that so figured I'd throw it in.  Wasn't fond that it brought an extra dependency up in the api package.

File: retrofit/src/main/java/org/devfleet/crest/retrofit/CrestServiceImpl.java [hunk 2]
```diff
@@ -269,6 +272,46 @@ private boolean setWaypoints(final List<CrestWaypoint> waypoints, final boolean
...
+
+            return dictionary.getItems();
+        } catch (IOException e) {
+            LOG.error(e.getLocalizedMessage(), e);
+            throw new IllegalStateException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    @Override
+    public List<CrestMarketBulkOrder> getAllOrders(final long regionId) {
```
Comment (evanova) [inline retrofit/src/main/java/org/devfleet/crest/retrofit/CrestServiceImpl.java]: Please rename to getAllMarketOrders for consistency

File: api/src/main/java/org/devfleet/crest/CrestService.java [hunk 3]
```diff
@@ -42,4 +45,7 @@
 
     List<CrestMarketHistory> getMarketHistory(final long regionId, final long itemId);
 
+    List<CrestMarketOrder> getMarketOrders(final long regionId, final String orderType, final long itemId);
+
+    List<CrestMarketBulkOrder> getAllOrders(final long regionId );
```
Comment (evanova) [inline api/src/main/java/org/devfleet/crest/CrestService.java]: Please rename to getAllMarketOrders for consistency

**Comment by evanova** [comment]
Just a few more renaming (see my comments) and we're good to go.
Thanks

**Pull request merged**